src/main/java/org/example/controllers/ApartmentController.java

```diff
@@ -0,0 +1,54 @@
+package org.example.controllers;
+
+import org.example.dto.ApartmentDTO;
+import org.example.service.ApartmentService;
+import org.example.service.RealtyAgentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+public class ApartmentController {
+
+    @Autowired
+    protected ApartmentService apartmentService;
+
+    @Autowired
+    protected RealtyAgentService realtyAgentService;
+
+    @GetMapping("/apartmentlist")
+    public String showApartments(@RequestParam("agent.id") int agentId, ModelMap modelMap) {
+        modelMap.put("agent", realtyAgentService.findRealtyAgentById(agentId));
+        modelMap.put("apartments", apartmentService.findAllApartmentsByRealtyAgent(agentId));
+        return "/apartmentlist";
+    }
+
+    @GetMapping("/apartmentadd")
+    public String addApartment(@RequestParam("agent.id") int agentId, ModelMap modelMap) {
+        modelMap.put("agent", realtyAgentService.findRealtyAgentById(agentId));
+        modelMap.addAttribute("apartment", new ApartmentDTO());
+        return "/apartmentadd";
+    }
+
+    @PostMapping("/apartmentadd")
+    public String saveApartment(ModelMap modelMap, ApartmentDTO apartmentDTO) {
+        modelMap.put("apartment", apartmentService.saveApartment(apartmentDTO));
```

Comment:
You don't need to put it in the model, since you've done redirect here, which ignore all parameters, passed to the model